### PR TITLE
Improvement/docker testing workflow branch

### PR DESCRIPTION
**Pull request by snova-kwasia**

### CODE REVIEW

File: .github/workflows/docker-build-and-test.yml [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
 
 jobs:
   build_and_test:
-    if: ${{ github.ref == 'refs/heads/improvement/docker-testing-workflow-branch' || github.event_name == 'workflow_dispatch' }} #TODO: Remove
     runs-on: ubuntu-latest
     env:
-      DOCKER_IMAGE: aisk-unit-tests  # Updated Docker image name
-      DOCKER_TAG: ${{ github.sha }}
+      DOCKER_IMAGE: aisk-unit-tests  # Docker image name
+      DOCKER_TAG: ${{ github.sha }}   # Git commit hash as the Docker tag
```
Comment (snova-francescar) [inline .github/workflows/docker-build-and-test.yml:25]: I don't understand the comment

File: tests/test_framework.py [hunk 2]
```diff
@@ -21,16 +21,21 @@
...
+import wandb  # Import wandb for Weights & Biases logging
 
 # Add the project root to the Python path
-project_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
-sys.path.insert(0, project_root)
+current_dir = os.path.dirname(os.path.abspath(__file__))
+repo_dir = os.path.abspath(os.path.join(current_dir, '..'))
+sys.path.append(repo_dir)
+
+from utils.visual.env_utils import get_wandb_key  # Import get_wandb_key
```
Comment (snova-francescar) [inline tests/test_framework.py:33]: Remove comment

File: tests/test_framework.py [hunk 3]
```diff
@@ -39,21 +44,32 @@
     'financial_assistant',
     'function_calling',
     'search_assistant',
-    'benchmarking',
     'image_search',
-    'multimodal_knowledge_retriever',
+    #'multimodal_knowledge_retriever',
```
Comment (snova-francescar) [inline tests/test_framework.py:48]: Add TODO to include this later?

File: tests/test_framework.py [hunk 4]
```diff
@@ -39,21 +44,32 @@
...
+        '--num-input-tokens 1000 '
+        '--num-output-tokens 1000 '
+        '--num-requests 2 '
+        '--llm-api sncloud'
+    ),
     'enterprise_knowledge_retriever': 'python tests/ekr_test.py',
     'financial_assistant': 'python tests/financial_assistant_test.py',
     'function_calling': 'python tests/fc_test.py',
-    'multimodal_knowledge_retriever': 'python tests/multimodal_knowledge_retriever_test.py',
+    # 'multimodal_knowledge_retriever': (
```
Comment (snova-francescar) [inline tests/test_framework.py:70]: Add TODO here.

File: .github/workflows/docker-build-and-test.yml (hunk 1 above)
Comment (snova-kwasia) [reply .github/workflows/docker-build-and-test.yml:25]: In Docker you label images with a tag, we are using the git commit hash as the label.

File: tests/test_framework.py (hunk 4 above)
Comment (snova-kwasia) [reply tests/test_framework.py:70]: Done

File: tests/test_framework.py (hunk 3 above)
Comment (snova-kwasia) [reply tests/test_framework.py:48]: Done

File: utils/visual/env_utils.py [hunk 5]
```diff
@@ -88,30 +88,38 @@ def save_credentials(api_key: str, additional_vars: Optional[Dict[str, Any]] = N
...
 
-def get_wandb_key() -> Optional[Any]:
+import netrc
+import os
+from typing import Optional
+
+
+def get_wandb_key() -> Optional[str]:
+    """
+    Retrieve the Weights & Biases API key from the environment or ~/.netrc,
```
Comment (snova-francescar) [inline utils/visual/env_utils.py:98]: What is `netrc`?

Comment (snova-kwasia) [reply utils/visual/env_utils.py:98]: The .netrc file contains login information used by the auto-login process. It's a standard used by a lot of companies including W&B

File: .github/workflows/docker-build-and-test.yml [hunk 6]
```diff
@@ -1,21 +1,28 @@
...
-        required: false
+  # Runs at every push on main
   push:
-    branches: [ "improvement/docker-testing-workflow-branch" ]  # Isolated branch name for testing to ensure we don't affect main
+    branches:
+      - main
+
+  # Runs at every pull request review
+  pull_request_review:
+    types: [submitted]
```
Comment (snova-francescar) [inline .github/workflows/docker-build-and-test.yml:11]: What happens if after the review has been submitted another push is made to the open PR?
We would have potentially breaking changes, that are not covered by this test.

Comment (snova-francescar) [reply .github/workflows/docker-build-and-test.yml:11]: Any possible problems will be caught right after merge, as every push to main triggers tests.

**Review by snova-jorgep — APPROVED** [review]

**Review by snova-francescar — APPROVED** [review]

**Pull request merged**